transport-native-unix-common/src/main/c/netty_unix_socket.c

```diff
@@ -183,6 +189,27 @@ static int socket_type(JNIEnv* env) {
         }
         return AF_INET6;
     } else {
+        int optval = 1;
+        if (netty_unix_socket_setOption0(fd, IPPROTO_IPV6, IPV6_V6ONLY, &optval, sizeof(optval)) < 0) {
```

Comment:
For the same input (ipv4Preferred)  yeah.  Like I said, I am not as familiar with native code here, but some things make me antsy.  For example, calling bind on the same Address below, won't that be racy if two sockets are created at almost the same time?  Woudn't that imply you need to do so_reuseaddr ?  If lots of connections came in quick, it would waste ports.  

Up to you, but as is I think it slows down reading; it isn't as "obviously correct" as it could be.